src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1452,7 +1452,10 @@ private static IEnumerable<TResult> RepeatIterator<TResult>(TResult element, int
 
         public static IEnumerable<TResult> Empty<TResult>()
         {
-            return EmptyEnumerable<TResult>.Instance;
+            // This is a more memory-intensive empty enumerable that allocates a new enumerator each time.
```

Comment:
As an aside, correct me if I am wrong, but it looks like `SZGenericArrayEnumerator<T>.Empty` is being greedily allocated as soon as `SZGenericArrayEnumerator<T>` is used for the first time (first time the generic `GetEnumerator` is called). Which means lots of possible unnecessary empty enumerator allocations for each T. Wouldn't it be better if this was lazily allocated only when an empty enumerator was actually needed? Something like `EmptySZGenericArrayEnumerator<T>.Instance`. (I can open an issue/PR on CoreCLR to continue the discussion there, if warranted).